File: src/Horse.Jockey/Models/Routers/AddBindingModel.cs
using System.Text.Json.Serialization;
using Newtonsoft.Json;

namespace Horse.Jockey.Models.Routers
{
    public class AddBindingModel
    {
        [JsonProperty("router")]
        [JsonPropertyName("router")]
        public string Router { get; set; }
        
        [JsonProperty("name")]
        [JsonPropertyName("name")]
        public string Name { get; set; }
        
        [JsonProperty("target")]
        [JsonPropertyName("target")]
        public string Target { get; set; }
        
        [JsonProperty("priority")]
        [JsonPropertyName("priority")]
        public int Priority { get; set; }
        
        [JsonProperty("type")]
        [JsonPropertyName("type")]
        public string Type { get; set; }
        
        [JsonProperty("interaction")]
        [JsonPropertyName("interaction")]
        public string Interaction { get; set; }
        
        [JsonProperty("contentType")]
        [JsonPropertyName("contentType")]
        public ushort? ContentType { get; set; }
        
        [JsonProperty("method")]
        [JsonPropertyName("method")]
        public string Method { get; set; }
    }
}

File: web/src/models/add-binding-model.ts

export class AddBindingModel {
    router: string | undefined;
    name: string | undefined;
    target: string | undefined;
    priority: number = 0;
    contentType: number | null = null;
    type: string | undefined;
    customBindingType: string | undefined;
    interaction: 'None' | 'Response' | undefined;
    method: 'RoundRobin' | 'Distribute' | 'OnlyFirst' | 'Get' | 'Post' | 'Put' | 'Patch' | 'Delete' | undefined;
}